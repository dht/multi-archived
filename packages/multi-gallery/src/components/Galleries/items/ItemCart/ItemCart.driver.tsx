import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ItemCart, ItemCartProps } from './ItemCart';
import { BaseComponentDriver } from 'testing-base';

export class ItemCartDriver extends BaseComponentDriver {
    private props: Partial<ItemCartProps> = {};

    constructor() {
        super('ItemCart');
    }

    when: any = {
        rendered: () => {
            render(<ItemCart {...(this.props as ItemCartProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(
                <ItemCart {...(this.props as ItemCartProps)} />
            );
        },
    };

    given: any = {
        props: (props: Partial<ItemCartProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        containerClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
