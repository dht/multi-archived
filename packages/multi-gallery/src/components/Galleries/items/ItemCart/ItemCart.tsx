import React from 'react';
import { Wrapper } from './ItemCart.style';

export type ItemCartProps = {};

export function ItemCart(_props: ItemCartProps) {
    return (
        <Wrapper className='ItemCart-wrapper' data-testid='ItemCart-wrapper'>
            ItemCart
        </Wrapper>
    );
}

export default ItemCart;
