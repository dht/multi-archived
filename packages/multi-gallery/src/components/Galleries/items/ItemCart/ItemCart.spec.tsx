import { ItemCartDriver } from './ItemCart.driver';
import Chance from 'chance';

const chance = new Chance();

describe('ItemCart', () => {
    let driver: ItemCartDriver;

    beforeAll(() => {
        driver = new ItemCartDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const containerClassName = element.get.containerClassName();
        const innerText = element.get.label();

        expect(containerClassName).toContain('ItemCart-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('ItemCart snapshots', () => {
    let driver: ItemCartDriver;

    beforeAll(() => {
        driver = new ItemCartDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
