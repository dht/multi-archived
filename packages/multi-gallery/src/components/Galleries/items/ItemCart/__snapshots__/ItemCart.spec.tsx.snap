// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCart snapshots should match snapshot 1`] = `
<div
  className="sc-bczRLJ WtYQR ItemCart-wrapper"
  data-testid="ItemCart-wrapper"
>
  ItemCart
</div>
`;
